slackbridge/messages.py

```diff
@@ -127,20 +127,30 @@ def _post_to_fluffy(self, channel_name, user_bot, file_data):
         # it is streamed.
         r.raw.decode_content = True
         r = requests.post(
-            FILEHOST + '/upload',
+            FILEHOST + '/upload?json',
             files={'file': (filename, r.raw)},
-            allow_redirects=False,
         )
-        if r.status_code not in (301, 302):
+        if r.status_code != 200:
             log.err('Failed to upload (status code {}):'.format(
                 r.status_code,
             ))
             return
 
+        location = ''
+        resp = r.json()
+        if resp['success']:
+            if (file_data['filetype'] == 'text' and
+                    resp['uploaded_files'][filename]['paste']):
+                location = resp['uploaded_files'][filename]['paste']
+            else:
+                location = resp['uploaded_files'][filename]['raw']
```

Comment:
you could probably replace this if-statement with just:

```python
upload = resp['uploaded_files'][filename]
location = upload.get('paste', upload['raw'])
```

(assuming you are okay with trusting fluffy to detect what is text and what isn't)

if you keep as-is, i'd be worried about the `and resp['uploaded_files'][filename]['paste']` in your `if` statement; this will raise a KeyError in cases where you think it's text but for whatever reason fluffy doesn't think it is (the `and resp['uploaded_files'][filename]['paste']` doesn't really make sense since the fluffy api never returns `"paste": null`, it just omits the key)